### ISSUE TITLE
[FEATURE] ability to specify revision method during predict

### ISSUE THREAD

**Issue by ryanvolpi** [carlomazzaferro/scikit-hts#28]
**Is your feature request related to a problem? Please describe.**
If I want to create predictions with multiple different revision methods I need to repeatedly start over, define a new regressor, and retrain the all of the models. 

**Describe the solution like**
The revision method only comes into play during the prediction stage and does not alter the training step whatsoever. Therefore, I think it should be an option to pass the revision method name to the predict function which will supercede the method specified to the htsregressor object. 

**Describe alternatives you've considered**
You can set model.revision_method.name = 'OLS' before calling model.predict( ) to change what revision method is used. I don't think this is the best way to do it. 

**Additional context**
Let me know your thoughts. I can implement this ASAP. Should a method be specified when instantiating the regressor object at all?

**Comment by mpearmain** [comment]
From an interested users perspective, (who is keen to get involved) this decoupling of the revision from the training makes sense to me, however as a strong decoupling 

This feels very hacky to me: 
_"being able to pass the revision method name to the `predict` function which will supercede the method specified to the HTSRegressor object"_
 
This kinda breaks the point of having a revision method in the `__init__` 

To me, the `HTSRegressor` is more a syntactic sugar on a full and known pipeline of operations and not investigation class so to speak. 

Personally, as someone who wants to be a power user of the library, i would like to solve using the lower level classes and 'build' my own custom setup by using the classes. (sorry im not familiar with underlying classes but psudeo code might be )

> clf = HTSRegressor(model='prophet', revision_method='OLS')`

can decompose into:
```python
from hts.hierarchy import HierarchyTree
ht = {**total, **state_h, **store_h, **dept_h, **item_h}  # Taken from M5 example
hierarchy = HierarchyTree.from_nodes(nodes=ht, df=df)

base_clf = HTSBaseRegressor(model='prophet')
base_clf.fit(df, hierarchy)
preds = base_clf.predict(...)

revision1 = HTSRevisionRegressor(model='OLS')
revision1.fit(preds, hierarchy)

revision2 = HTSRevisionRegressor(model='WLSS')
revision2.fit(preds, hierarchy)
```

or does the following solve the issue (which feels a bit weird calling a `convenience` function)

```python
clf = HTSRegressor(model='prophet', revision_method='None')
clf.fit(df, hierarchy)
preds = clf.pred(...)
revision1 = clf.convenience.revise_forecasts(preds, model='OLS')
revision2 = clf.convenience.revise_forecasts(preds, model='WLSS')
```

In the first example, it lets me bring my own predictions to the `HTSRevisionRegressor` and is agnostic to the models or pipelines used to generate them

**Comment by ryanvolpi** [comment]
That is a good point. I like the first method you suggest as it seems in line with the sklearn API. Also, it is similar to the functionality of R's HTS package. What do you think @carlomazzaferro?

**Comment by carlomazzaferro** [comment]
@mpearmain you got it completely right that the HTSRegressor is simply a pipelining operation between your usual forecasting and the application of the revision methodologies. Something that I've added that probably I should've done a better job documenting is the `hts.convenience.revise_forecasts` method:  https://scikit-hts.readthedocs.io/en/latest/modules.html#module-hts.convenience

You can bring your own forecasts (provided residuals, errors forecasts _or_ summing matrix for OLS, see docstrings for what you need before calling), and use the revision at your own leisure. FWIW, I'd be more inclined for a scikit-like implementation of the convenience function, both for the sake of consistency of the goal of the library and for developer experience. 

While I'm quite backed up recently, I'd be happy to discuss implementation details with anyone willing to take that effort over. Otherwise, let me know if there's any clarification needed as far as using the convenience function.

**Comment by mpearmain** [comment]
Id be happy to help in that effort, (like everyone im backed up as well) 
As you state the real power of this library is about resolving the hierarchical nature of forecasts and the revision there after.

With this effort i'll also try and add an HTS-example using some of the auto-forecasting packages like https://github.com/firmai/atspy to show it working external to the `HTSRegressor` class and taking advantage of the refactored class `HTSRevsionRegressor` ?

**Comment by mpearmain** [comment]
@carlomazzaferro did you want to discuss this request further in terms of design?

**Comment by carlomazzaferro** [comment]
@mpearmain I'm sorry I dropped out of the conversation for some time -- work and personal life have not been the nicest on my side recently, but I've found some renewed interest in working on this. If you are still interested in contributing, I'm open to your proposed solution

**Comment by carlomazzaferro** [comment]
Closing due to inactivity

**Comment by karta282950** [comment]
Hi, I found the issue recently.How can i use revise predict result by hts after i train a xgboost model?thans~
